pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -163,14 +163,77 @@ in
     nvimRequireCheck = "autosave";
   };
 
-  avante-nvim = (callPackage ./avante-nvim { }).overrideAttrs {
-    dependencies = with self; [
-      dressing-nvim
-      nui-nvim
-      nvim-treesitter
-      plenary-nvim
-    ];
-  };
+  avante-nvim =
+    let
+      version = "2024-11-03";
+      src = fetchFromGitHub {
+        owner = "yetone";
+        repo = "avante.nvim";
+        rev = "58e0165098c8a4d4c832fd31e3d613a5c4b28a35";
+        hash = "sha256-E42IsRph1TFnuf/rE+0TAD25KkwWAeWI0UNcTpEn3ms=";
+      };
+      meta = with lib; {
+        description = "Neovim plugin designed to emulate the behaviour of the Cursor AI IDE";
+        homepage = "https://github.com/yetone/avante.nvim";
+        license = licenses.asl20;
+        maintainers = with lib.maintainers; [
+          ttrei
+          aarnphm
+        ];
+      };
+      avante-nvim-lib = rustPlatform.buildRustPackage {
+        pname = "avante-nvim-lib";
+        inherit version src meta;
+
+        cargoHash = "sha256-QLAPgZaeuTGwfeAgh6EtDbduM6TvZMXg1n2k6mguGVI=";
+
+        nativeBuildInputs = [
+          pkg-config
+        ];
+
+        buildInputs =
+          [
+            openssl
+          ]
+          ++ lib.optionals stdenv.hostPlatform.isDarwin [
+            darwin.apple_sdk.frameworks.Security
+          ];
```

Comment:
You can probably get rid of this now as the darwin refactor has landed in `master`.

Reply:
Yeah, if we're going to touch all of the code for this plugin, could you remove this optional? It will automatically get all apple-sdk frameworks in the stdenv, now. I'm not familiar enough with their dependencies to know if it could benefit from a higher level SDK. But, I do see they are targeting macos-latest in their CI runner.